### PR TITLE
Worker versioning - add BuildIDs search attribute

### PR DESCRIPTION
**Pull request by bergundy**
Part of #3303

### CODE REVIEW

**Review by rodrigozhou — APPROVED** [review]
The part adding the search attribute looks good to me.

File: schema/elasticsearch/visibility/versioned/v5/index_template_v7.json
Comment (bergundy) [reply schema/elasticsearch/visibility/versioned/v5/index_template_v7.json:81]: Good point, I thought about that but because it's plural I think it'd be clearer if IDs stands out more.
I'm now not so convinced.
I'll get another opinion.

Comment (alexshtin) [inline schema/elasticsearch/visibility/versioned/v5/index_template_v7.json:81]:
```suggestion
      "BuildIds": {
```

Comment (alexshtin) [reply schema/elasticsearch/visibility/versioned/v5/index_template_v7.json:81]: Here you are.

File: service/history/workflow/workflow_task_state_machine.go
Comment (dnr) [inline service/history/workflow/workflow_task_state_machine.go:541]: can we/should we do anything about this? are we going to have these repeated twice in the event for now?

File: service/matching/matchingEngine.go
Comment (dnr) [inline service/matching/matchingEngine.go:1190]: don't do this here, instead, leave the build stamp out of the AddWorkflowTaskRequest on the history side (there's no reason for matching to know about it if it's not going to use it)

File: tests/integrationbase.go
Comment (dnr) [inline tests/integrationbase.go:377]: use `s.Eventually`

File: service/history/workflow/mutable_state_impl.go
Comment (bergundy) [reply service/history/workflow/mutable_state_impl.go:1905]: It can't here because if we don't add a reset point the function returns early

Comment (alexshtin) [reply service/history/workflow/mutable_state_impl.go:1905]:
```suggestion
	resetPoints := exeInfo.GetAutoResetPoints().GetPoints()
```
is guaranteed to be safe.

**Review by alexshtin — APPROVED** [review]
LGTM, but I do insist on `BuildIDs` to `BuildIds` rename everywhere besides go code. I personally would do it everywhere in go code too, but because of old go linter (which we don't currently use) all our `Id` are `ID` in go :-(.

File: service/history/workflow/workflow_task_state_machine.go
Comment (bergundy) [reply service/history/workflow/workflow_task_state_machine.go:541]: Hmm... I think the worker will send both, let me track this and figure out what we want to do.

File: service/matching/matchingEngine.go
Comment (bergundy) [reply service/matching/matchingEngine.go:1190]: I think we'll need to to not send the stamp directly from history as I mentioned in my review of your last PR.
For now I'm leaving this on so the tests pass.

Comment (bergundy) [reply service/matching/matchingEngine.go:1190]: Let's fix before we merge and this of course depends on your changes too.

File: tests/integrationbase.go
Comment (bergundy) [reply tests/integrationbase.go:377]: Thanks, wasn't aware of that.

**Comment by bergundy** [comment]
Ended up renaming `BuildID` to `BuildId` in the entire code base.
I'll address some of the open comments in a follow up PR.

**Pull request merged**
